command/validate.py

```diff
@@ -1,8 +1,239 @@
+import logging
+import json
+
+logger = logging.getLogger(__name__)
+
 class ValidateCmd():
-    name = 'validate'
+    dataKey = "data"
+    crdKey = "customResourceDefinitions"
+    csvKey = "clusterServiceVersions"
+    pkgsKey = "packages"
 
     def __init__(self):
         pass
         
-    def validate(self):
-        print("Hello, world. The validate feature is currently under construction.")
+    def validate(self, bundle):
+        """validate takes a bundle as a dictionary and returns a boolean value that
+        describes if the bundle is valid. It also logs verification information when
+        there are invalid sections in the bundle.
+
+        :param bundle: Dictionary of bundle value
+        """
+        logger.info("Validating bundle.")
+
+        return self._bundle_validation(bundle)
+
+    def _bundle_validation(self, bundle):
+        validationDict = dict()
+
+        if self.dataKey not in bundle:
+            logger.error("Bundle does not contain base data field.")
```

Comment:
In L8 we refer to the CSV/CRD/Package as Keys yet here we refer to them as fields. Should we converge on a single naming convention?

Reply:
@awgreene the word `key` is used here to describe the mechanical feature of key,value in the `dict` type. Whenever the code refers to `key` it is describing a hardcoded string that is used to extract specific data. Field is used to describe the concept of a generic data field on a type, which is a different concept. They aren't interchangeable, so I am using different words here.

Reply:
I recognize the distinction and agree that using the term `field` would be easier for users to absorb. LGTM.